src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/model/RoleFilter.kt

```diff
@@ -1,64 +1,167 @@
 package uk.gov.justice.digital.hmpps.externalusersapi.model
 
-import com.google.common.collect.ImmutableList
-import org.springframework.data.jpa.domain.Specification
-import javax.persistence.criteria.CriteriaBuilder
-import javax.persistence.criteria.CriteriaQuery
-import javax.persistence.criteria.Path
-import javax.persistence.criteria.Predicate
-import javax.persistence.criteria.Root
-import kotlin.reflect.KProperty1
+import org.springframework.data.domain.Pageable
+import java.lang.String.format
+import java.util.Optional
+import java.util.regex.Pattern
+
+private const val COUNT_SQL =
+  """
+    SELECT COUNT(*) AS roleCount FROM (%s) AS allRoles
+  """
+
+private const val PROJECTION =
+  """
+  SELECT
+  DISTINCT r.role_id,
+           r.admin_type,
+           r.role_code,
+           r.role_description,
+           r.role_name 
+  FROM roles r 
+  """
+
+private const val FILTER_START =
+  """
+    WHERE
+  """
+
+private const val SUBSEQUENT_FILTER =
+  """
+    AND 
+  """
+
+private const val ROLE_NAME_FILTER =
+  """
+    (LOWER(r.role_name) like '%s') 
+  """
```

Comment:
Could our role name come in as "This is my name" - so do we need to lower both sides?